File: app/src/main/java/dev/ragnarok/fenrir/api/services/INotificationsService.java
package dev.ragnarok.fenrir.api.services;

import dev.ragnarok.fenrir.api.model.response.BaseResponse;
import dev.ragnarok.fenrir.api.model.response.NotificationsResponse;
import dev.ragnarok.fenrir.model.AnswerVKOfficialList;
import io.reactivex.rxjava3.core.Single;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

public interface INotificationsService {

    @POST("notifications.markAsViewed")
    Single<BaseResponse<Integer>> markAsViewed();

    @FormUrlEncoded
    @POST("notifications.get")
    Single<BaseResponse<NotificationsResponse>> get(@Field("count") Integer count,
                                                    @Field("start_from") String startFrom,
                                                    @Field("filters") String filters,
                                                    @Field("start_time") Long startTime,
                                                    @Field("end_time") Long endTime);

    @FormUrlEncoded
    @POST("notifications.get")
    Single<BaseResponse<AnswerVKOfficialList>> getOfficial(@Field("count") Integer count,
                                                           @Field("start_from") Integer startFrom,
                                                           @Field("filters") String filters,
                                                           @Field("start_time") Long startTime,
                                                           @Field("end_time") Long endTime,
                                                           @Field("fields") String fields);

}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/api/services/INotificationsService.kt
package dev.ragnarok.fenrir.api.services

import dev.ragnarok.fenrir.api.model.response.BaseResponse
import dev.ragnarok.fenrir.api.model.response.NotificationsResponse
import dev.ragnarok.fenrir.api.rest.IServiceRest
import dev.ragnarok.fenrir.model.FeedbackVKOfficialList
import kotlinx.coroutines.flow.Flow

class INotificationsService : IServiceRest() {
    val markAsViewed: Flow<BaseResponse<Int>>
        get() = rest.request("notifications.markAsViewed", null, baseInt)

    operator fun get(
        count: Int?,
        startFrom: String?,
        filters: String?
    ): Flow<BaseResponse<NotificationsResponse>> {
        return rest.request(
            "notifications.get", form(
                "count" to count,
                "start_from" to startFrom,
                "filters" to filters,
            ), base(NotificationsResponse.serializer())
        )
    }

    fun getOfficial(
        count: Int?,
        startFrom: Int?,
        fields: String?
    ): Flow<BaseResponse<FeedbackVKOfficialList>> {
        return rest.request(
            "notifications.get", form(
                "count" to count,
                "start_from" to startFrom,
                "fields" to fields
            ), base(FeedbackVKOfficialList.serializer())
        )
    }

    fun hide(query: String?): Flow<BaseResponse<Int>> {
        return rest.request("notifications.hide", form("query" to query), baseInt)
    }
}